File: frontend/davon-library-webui/src/lib/types/book.ts
export interface Book {
    id: string;
    title: string;
    author: string;
    isbn: string;
    category: string;
    description: string;
    publishDate: string;
    status: 'available' | 'borrowed' | 'maintenance';
    coverImage?: string;
    createdAt: string;
    updatedAt: string;
}

export interface BookCreateInput {
    title: string;
    author: string;
    isbn: string;
    category: string;
    description: string;
    publishDate: string;
    status: 'available' | 'borrowed' | 'maintenance';
    coverImage?: string;
}

export interface BookUpdateInput {
    title?: string;
    author?: string;
    isbn?: string;
    category?: string;
    description?: string;
    publishDate?: string;
    status?: 'available' | 'borrowed' | 'maintenance';
    coverImage?: string;
} 

File: backend/davon-library-backend/src/main/java/org/acme/model/Book.java
package org.acme.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "books")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    private String isbn;
    private String description;
    private LocalDate publishDate;
    private String coverImage;
    @Enumerated(EnumType.STRING)
    private BookStatus status;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id", nullable = false)
    private Author author;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", nullable = false)
    private Category category;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    // Getters and setters
}